test/dirichlet.jl

```diff
@@ -127,3 +128,54 @@ end
     @test entropy(Dirichlet(N, 1)) ≈ -loggamma(N)
     @test entropy(Dirichlet(ones(N))) ≈ -loggamma(N)
 end
+
+@testset "Dirichlet differentiation $n" for n in (2, 10)
+    alpha = rand(n)
+    Δalpha = randn(n)
+    d, ∂d = ChainRulesCore.frule((nothing, Δalpha), Dirichlet, alpha)
+    ChainRulesTestUtils.test_frule(Dirichlet ⊢ ChainRulesCore.NoTangent(), alpha ⊢ Δalpha)
+    _, dp = ChainRulesCore.rrule(Dirichlet, alpha)
+    ChainRulesTestUtils.test_rrule(Dirichlet{Float64} ⊢ ChainRulesCore.NoTangent(), alpha)
+    x = rand(n)
+    x ./= sum(x)
+    Δx = 0.05 * rand(n)
+    Δx .-= mean(Δx)
+    # such that x ∈ Δ, x + Δx ∈ Δ
```

Comment:
Is the restriction on `Δx` needed? It seems unnecessary for the `test_frule` below.

Reply:
thought so too but otherwise your second point is outside the domain with probability 1
```
  Expression: isapprox(actual, expected; kwargs...)
   Evaluated: isapprox(1.0589089695661067, NaN; rtol = 1.0e-9, atol = 1.0e-9)
```

Reply:
this is also the case with fdm set to forward